### ISSUE TITLE
Weird bugs that strangely don't get fixed

### ISSUE THREAD

**Issue by PinkSheep752** [DS-Homebrew/TWiLightMenu#2455]
Okay, so when I put my microSD card into my New 3DS XL... everything seems to work fine, doesn't it?

But when I launch the SELECT menu and launch the manual, most pages are colored VERY incorrectly.

I have tried everything from formatting the microSD card SEVERAL times, to using TWLFix, to reinstalling TWiLightMenu++ entirely... but nothing seemed to fix it.

Also, I can't really describe it but some of the music in the application just didn't sound right to me.

And finally, when I tried playing a translation for Puyo Puyo!! 20th Anniversary, the patch got really glitched out to the point where so many things weren't right and the nds-bootstrap menu would just crash when I try to reset the game. The only way out of it was to press the HOME button and return to the 3DS HOME menu.

Base Console: I am using a New 3DS XL on the latest firmware version, along with the latest LumaCFW version.

TWiLight Menu++ Version: I am always using the latest base versions from Universal Updater.

TWiLight Menu++ Theme: Nintendo DSi, no skins.

Expected behavior: The manual should have a dark gray background and look normal throughout every single page.

Actual behavior: Most pages end up being incorrectly colored and having a sort-of corrupted look to them.

Steps to reproduce (it might be hard for others to reproduce it, but here's what happens on my end)

1. Format microSD using the SD format tool
2. Download necessary homebrew files and place them on the microSD Card
3. Plug microSD back into the 3DS
4. Download TWLMenu++ from Universal Updater and say yes to extracting the roms folder
5. Launch the sort-of corrupted TWiLight Menu++

Notes:

I honestly don't have any apart from the fact that I am CERTAIN my microSD card is not at all corrupted, as all my devices read the contents of it just fine.

**Comment by RocketRobz** [comment]
The manual colors being bugged is a known issue, though it's probably not an issue with TWLMenu++ itself, but an issue with the manual page image generator.

As for your microSD card, which brand is it?

**Comment by PinkSheep752** [comment]
> The manual colors being bugged is a known issue, though it's probably not an issue with TWLMenu++ itself, but an issue with the manual page image generator.
> 
> 
> 
> As for your microSD card, which brand is it?

Fair enough... but as for my microSD card, I am using a 32GB Micro Center branded microSD card. I got two of them at the same time and I am using the other one on my Wii U. Right now the contents of the one for the Wii U works perfectly so I don't see why it wouldn't work great on my New 3DS XL as well...

**Comment by PinkSheep752** [comment]
> > The manual colors being bugged is a known issue, though it's probably not an issue with TWLMenu++ itself, but an issue with the manual page image generator.
> > 
> > 
> > 
> > As for your microSD card, which brand is it?
> 
> Fair enough... but as for my microSD card, I am using a 32GB Micro Center branded microSD card. I got two of them at the same time and I am using the other one on my Wii U. Right now the contents of the one for the Wii U works perfectly so I don't see why it wouldn't work great on my New 3DS XL as well...

And by "two of them at the same time", I mean I got two 32GB microSD cards for one collective price.

**Comment by RocketRobz** [comment]
> but as for my microSD card, I am using a 32GB Micro Center branded microSD card. I got two of them at the same time and I am using the other one on my Wii U. Right now the contents of the one for the Wii U works perfectly so I don't see why it wouldn't work great on my New 3DS XL as well...

libfat (used by many DS/DSi homebrew) is known to not be stable with SD cards of unfamiliar brands.
It is recommended to instead try an SD card of a well-known brand, such as Kingston, Samsung, SanDisk, or Sony.

**Comment by PinkSheep752** [comment]
I also have a 4GB microSD card that came with the system when I got it... and that one's a different brand

**Comment by PinkSheep752** [comment]
Also, I have a Sandisk 32GB MicroSD card that is currently being used for my unmodded Switch.

**Comment by PinkSheep752** [comment]
Also, isn't there an alternative for libfat that makes compatibility with lesser known brands easier?

**Comment by RocketRobz** [comment]
> Also, isn't there an alternative for libfat that makes compatibility with lesser known brands easier?

There is libslim (which uses FatFS), though it's slightly bugged (for example, it cannot correctly get the SD cluster size).

**Comment by PinkSheep752** [comment]
> > Also, isn't there an alternative for libfat that makes compatibility with lesser known brands easier?
> 
> 
> 
> There is libslim (which uses FatFS), though it's slightly bugged (for example, it cannot correctly get the SD cluster size).

How often does libfat and/or libslim get updated? If they're basically dead in the water then there should be something else we can use instead, as to not screw up compatibility with SD cards and USB devices

**Comment by RocketRobz** [comment]
> How often does libfat and/or libslim get updated? If they're basically dead in the water then there should be something else we can use instead, as to not screw up compatibility with SD cards and USB devices

afaik, they do not get updated nowadays, and there are no alternatives to them.
